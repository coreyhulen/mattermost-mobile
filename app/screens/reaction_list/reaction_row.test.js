// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React from 'react';
import {shallow} from 'enzyme';

import ReactionRow from './reaction_row';

describe('ReactionRow', () => {
    const baseProps = {
        emojiName: 'smile',
        navigator: {},
        teammateNameDisplay: 'username',
        theme: {
            centerChannelColor: '#aaa',
        },
        user: {id: 'user_id', username: 'username'},
    };

    test('should match snapshot, renderContent', () => {
        const wrapper = shallow(
            <ReactionRow {...baseProps}/>
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
