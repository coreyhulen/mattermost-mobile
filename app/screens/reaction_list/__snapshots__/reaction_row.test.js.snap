// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactionRow should match snapshot, renderContent 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": undefined,
      "flexDirection": "row",
      "height": 44,
      "justifyContent": "flex-start",
      "width": "100%",
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "width": "15%",
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Connect(ProfilePicture)
        showStatus={false}
        size={24}
        userId="user_id"
      />
    </TouchableOpacity>
  </Component>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Object {
        "flexDirection": "row",
        "marginLeft": 5,
        "width": "70%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#aaa",
          "fontSize": 14,
          "paddingRight": 5,
        }
      }
    >
      @username
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
        
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "rgba(170,170,170,0.5)",
          "fontSize": 14,
          "marginLeft": 5,
        }
      }
    >
      username
    </Text>
  </Text>
  <Component
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "width": "15%",
      }
    }
  >
    <Connect(Emoji)
      emojiName="smile"
      size={26}
    />
  </Component>
</Component>
`;
